pulp_puppet_plugins/pulp_puppet/plugins/importers/upload.py

```diff
@@ -38,7 +38,11 @@ def handle_uploaded_unit(repo, type_id, unit_key, metadata, file_path, conduit):
     extracted_data = metadata_parser.extract_metadata(file_path, repo.working_dir)
 
     # rename the file so it has the original module name
-    original_filename = extracted_data['name'] + '-' + extracted_data['version'] + '.tar.gz'
+    # since module's name information is taken from metadata.json, name can be
+    # in forman author/name or author-name. Module's name should be adjusted
+    # before the upload path creation
+    original_filename = extracted_data['name'].replace('/', '-', 1) + '-' + \
+        extracted_data['version'] + '.tar.gz'
```

Comment:
Why not reuse this method:

https://git.io/vrAtb

Like so:

https://gist.github.com/a507ea77b59dbbc3f4ecdf7505d4cf82